src/base/TransactionQueue.ts

```diff
@@ -213,10 +224,34 @@ export class TransactionQueue<
   public onTransactionStatusChange<TxArgs extends unknown[], Values extends unknown[]>(
     listener: (transaction: PolymeshTransaction<TxArgs, Values>, transactionQueue: this) => void
   ): () => void {
-    this.emitter.on(Events.TransactionStatusChange, listener);
+    this.emitter.on(Event.TransactionStatusChange, listener);
+
+    return (): void => {
+      this.emitter.removeListener(Event.TransactionStatusChange, listener);
+    };
+  }
+
+  /**
+   * Subscribe to the results of this queue being processed by the harvester (and as such, available to the middleware)
+   *
+   * @param listener - callback function that will be called whenever the middleware is updated with the latest data.
+   *   If there is an error (timeout or middleware offline) it will be passed to this callback
+   *
+   * @returns unsubscribe function
+   * @throws if the middleware wasn't enabled when instantiating the SDK client
+   */
+  public onProcessedByMiddleware(listener: (err?: PolymeshError) => void): () => void {
+    if (!this.context.isMiddlewareEnabled()) {
+      throw new PolymeshError({
+        code: ErrorCode.FatalError,
```

Comment:
I'm not sure if `FatalError` is the best option here. Why not `MiddlewareError`?

Reply:
It's not actually the middleware that is failing here. It's just a user error (akin to passing a string to a function that requires a number) so I figured fatal error is a better fit